src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx

```diff
@@ -101,16 +108,18 @@ export class EditDataSourceForm extends React.Component<EditDataSourceProps, Edi
   setFormValuesForEditMode() {
     if (this.props.existingDataSource) {
       const { title, description, endpoint, auth } = this.props.existingDataSource;
-
       this.setState({
         title,
         description: description || '',
         endpoint,
         auth: {
           type: auth.type,
           credentials: {
-            username: auth.type === AuthType.NoAuth ? '' : auth.credentials?.username || '',
-            password: auth.type === AuthType.NoAuth ? '' : this.maskedPassword,
+            username: auth.type === AuthType.UsernamePasswordType ? auth.credentials?.username : '',
+            password: auth.type === AuthType.UsernamePasswordType ? this.maskedPassword : '',
+            region: auth.type === AuthType.SigV4 ? auth.credentials?.region : '',
```

Comment:
maybe store the comparing result in a variable and reuse on line 120-122

Reply:
updated